onetl/connection/db_connection/jdbc_mixin/connection.py

```diff
@@ -308,6 +310,16 @@ def execute(
             log_lines(log, str(metrics))
         return df
 
+    @property
+    def jdbc_dialect(self):
```

Comment:
My proposal is:
* Implement `JDBCMixin._get_spark_dialect() method which will return Java object
* Rename `JDBCConnection.jdbc_dialect()` to `JDBCConnection._get_spark_dialect_name()`, to avoid confusion with [JDBCDialect](https://github.com/MobileTeleSystems/onetl/blob/1382600ec8552eb22bc53e242f1afea1ac38ab01/onetl/connection/db_connection/jdbc_connection/dialect.py#L18) class, and make it a method instead of property
* Call `self._get_spark_dialect()` here https://github.com/MobileTeleSystems/onetl/blob/1382600ec8552eb22bc53e242f1afea1ac38ab01/onetl/connection/db_connection/jdbc_mixin/connection.py#L562-L563 and in `JDBCConnection._get_spark_dialect_name()`

Reply:
3 item is not fully resolved

Reply:
if we split `_get_spark_dialect_name()` to `JDBCConnection` and `_get_spark_dialect()` to `JDBCMixin` then we would get  inconsistanced log messages between `.sql`, `.get_df_schema` and `.execute`, `.fetch` methods as they are splitted between `JDBCConnection` and  `JDBCMixin`

Reply:
Could you give an example?

Reply:
we have two options:
1.  in `.fetch`, `.execute` methods pass:
`log.info("|%s| Detected dialect: '%s'", self.__class__.__name__, self._get_spark_dialect_name())  #type: ignore`, otherwise mypy would fail with:

```
onetl/connection/db_connection/jdbc_mixin/connection.py:208: error: "JDBCMixin" has no attribute "_get_spark_dialect_name"; maybe "_get_spark_dialect"?  [attr-defined]
onetl/connection/db_connection/jdbc_mixin/connection.py:281: error: "JDBCMixin" has no attribute "_get_spark_dialect_name"; maybe "_get_spark_dialect"?  [attr-defined]

```
2.  `log.info("|%s| Detected dialect: '%s'", self.__class__.__name__, self._get_spark_dialect())` and log message would change to:
`|Postgres| Detected dialect: 'org.apache.spark.sql.jdbc.PostgresDialect$@485cc3e'`

Reply:
Just move both `_get_spark_dialect_name` and `_get_spark_dialect` to JdbcMixin

Reply:
it's already done

> 3 item is not fully resolved
I thought it was about methods location

Reply:
No, it was about calling `jdbc_dialects_package.JdbcDialects.get` twice in 2 different methods